### DISCUSSION TITLE
Question about _dual_steps definition.

### DISCUSSION THREAD

**Discussion by MKAbdElrahman** [flexcompute/tidy3d#193] [Q&A]
Since the boundary condition are periodic, the first center point is linked to the last center. 

Why you needed to subtract a more additional step ``` grid_size[dim] ```  in ```  centers[-1] - grid_size[dim] ```
```python
 def _dual_steps(self) -> Coords:
        """Return dual steps of the cells in the :class:`Grid`.
        Returns
        -------
        :class:`Coords`
            Distances between each of the cell centers along each dimension.
        """

        # We need the dual steps to be same size as grid.centers, so we pad with periodicity applied
        grid_size = [(bounds[1] - bounds[0]) for bounds in self.boundaries.to_list]
        dsteps = {}
        for dim, (key, centers) in enumerate(self.centers.dict(exclude={TYPE_TAG_STR}).items()):
            centers_pad = np.concatenate(([centers[-1] - grid_size[dim]], centers))
            dsteps[key] = np.diff(centers_pad)

        return Coords(**dsteps)

```

**Comment by momchilmm** [comment]
Yeah, this should be fixed to use the periodicity properly, good catch!

**Reply by MKAbdElrahman** [reply]
thanks!

**Reply by MKAbdElrahman** [reply]
I think it is wrong to define a step size  this way. The distance  h will be same as the domain length in one direction and that would lead to a big  finite difference error.  The proper step size is (h1 + h_end)/2 where h1 and h_end are the first and last step sizes.

**Reply by momchil-flex** [reply]
Fixed in 600331524f1ac1e3253007fde85db184ac98e5d9, thanks!
